packages/experiments/src/components/Coachmark/Coachmark.tsx

```diff
@@ -0,0 +1,226 @@
+        >
+          <div className={ classNames.pulsingBeacon }></div>
+          <div
+            className={ classNames.translateAnimationContainer }
+            ref={ this._resolveRef('_translateAnimationContainer') }
+          >
+            <div
+              className={ classNames.scaleAnimationLayer }
+              ref={ this._resolveRef('_scaleAnimationContainer') }
+            >
+              <div
+                className={ classNames.rotateAnimationLayer }
+                ref={ this._resolveRef('_rotateAnimationContainer') }
+              >
+                <div
+                  className={ classNames.entityHost }
+                  ref={ this._resolveRef('_entityHost') }
+                >
+                  <div
+                    className={ classNames.entityInnerHost }
+                    ref={ this._resolveRef('_entityInnerHostElement') }
+                  >
+                    { children }
+                  </div>
+                </div>
+              </div>
+            </div>
+          </div>
+        </div>
+      </PositioningContainer>
+    );
+  }
+
+  public componentWillReceiveProps(newProps: ICoachmarkProps) {
+    if (this.props.isCollapsed && !newProps.isCollapsed) {
+      // The coachmark is about to open
+      this._openCoachmark();
+    }
+  }
+
+  public componentDidMount() {
+    // If we have already accessed the width and height
+    // We do not wan't to do it again.
+    if (this.state.isMeasuring) {
+      this._async.setTimeout(() => {
+        if ((this.state.entityInnerHostRect.width + this.state.entityInnerHostRect.width) === 0) {
+          this.setState({
+            isMeasuring: false,
+            entityInnerHostRect: {
+              width: this._entityInnerHostElement.offsetWidth,
+              height: this._entityInnerHostElement.offsetHeight
+            }
+          });
+
+          this.forceUpdate();
+        }
+
+        // Initialize element in proximity now that initial
+        // measurements have been taken
+        this._isElementInProximity(this._entityInnerHostElement);
+      }, 10);
+    }
+  }
+
+  private _openCoachmark() {
+    this.setState({
+      isCollapsed: false
+    });
+
+    this._translateAnimationContainer.addEventListener('animationstart', () => {
+      if (this.props.onAnimationBegin) {
+        this.props.onAnimationBegin();
+      }
+    });
+
+    this._translateAnimationContainer.addEventListener('animationend', () => {
+      if (this.props.onAnimationEnd) {
+        this.props.onAnimationEnd();
+      }
+    });
+  }
+
+  private _isElementInProximity(targetElement: HTMLElement, mouseProximityOffset: number = 0) {
+    /**
+     * An array of cached ids returned when setTimeout runs during
+     * the window resize event trigger.
+     */
+    let timeoutIds: number[] = [];
+
+    /**
+     * The target element the mouse would be in
+     * proximity to
+     */
+    let targetElementRect: ClientRect;
```

Comment:
Do getBoundingClientRect here as well, otherwise when the mouse move it's possible that it won't be set since there won't necessarily be a resize event before the mouse is moved.

Reply:
Good catch